pyiron_base/jobs/job/extension/executable.py

```diff
@@ -203,13 +203,10 @@ def executable_path(self, new_path):
         else:
             self.storage.mpi = False
 
-    def to_dict(self):
-        executable_dict = self._type_to_dict()
-        executable_storage_dict = self.storage._type_to_dict()
-        executable_storage_dict["READ_ONLY"] = self.storage._read_only
-        executable_storage_dict.update(self.storage.to_builtin())
-        executable_dict["executable"] = executable_storage_dict
-        return executable_dict
+    def _to_dict(self):
+        full_dict = self.storage.to_dict()
+        data = full_dict.pop("data")
+        return {"executable": {**full_dict, **data}}
```

Comment:
In principle, I like the idea with the `DummyHDFio`. But for classes like the `Executable`, I am not even sure if they need a `DataContainer`. I think the goal is to get to `pickle`-able classes in the future, with the `to_dict()` and `from_dict()` method being part of the transition. So I feel switching to `_to_dict()` here goes in the wrong direction.

Reply:
The DummyHDFio is not used for the executable or the data container, so I'm not sure what you mean.  I agree that in the future `Executable` could be a simple dataclass or so, but again here I just modified it, so it conforms to the new _to_dict.